examples/models/unet.py

```diff
@@ -0,0 +1,148 @@
+
+        self.stage3 = nn.Sequential(
+            SparseConvBlock(cs[2], cs[2], kernel_size=2, stride=2, dilation=1),
+            SparseResBlock(cs[2], cs[3], kernel_size=3, stride=1, dilation=1),
+            SparseResBlock(cs[3], cs[3], kernel_size=3, stride=1, dilation=1),
+        )
+
+        self.stage4 = nn.Sequential(
+            SparseConvBlock(cs[3], cs[3], kernel_size=2, stride=2, dilation=1),
+            SparseResBlock(cs[3], cs[4], kernel_size=3, stride=1, dilation=1),
+            SparseResBlock(cs[4], cs[4], kernel_size=3, stride=1, dilation=1),
+        )
+
+        self.up1 = nn.ModuleList([
+            SparseDeConvBlock(cs[4], cs[5], kernel_size=2, stride=2),
+            nn.Sequential(
+                SparseResBlock(cs[5] + cs[3],
+                               cs[5],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+                SparseResBlock(cs[5],
+                               cs[5],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+            )
+        ])
+
+        self.up2 = nn.ModuleList([
+            SparseDeConvBlock(cs[5], cs[6], kernel_size=2, stride=2),
+            nn.Sequential(
+                SparseResBlock(cs[6] + cs[2],
+                               cs[6],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+                SparseResBlock(cs[6],
+                               cs[6],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+            )
+        ])
+
+        self.up3 = nn.ModuleList([
+            SparseDeConvBlock(cs[6], cs[7], kernel_size=2, stride=2),
+            nn.Sequential(
+                SparseResBlock(cs[7] + cs[1],
+                               cs[7],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+                SparseResBlock(cs[7],
+                               cs[7],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+            )
+        ])
+
+        self.up4 = nn.ModuleList([
+            SparseDeConvBlock(cs[7], cs[8], kernel_size=2, stride=2),
+            nn.Sequential(
+                SparseResBlock(cs[8] + cs[0],
+                               cs[8],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+                SparseResBlock(cs[8],
+                               cs[8],
+                               kernel_size=3,
+                               stride=1,
+                               dilation=1),
+            )
+        ])
```

Comment:
This part is very repetitive. We could maybe use a for-loop for better code reuse.